src/polynomial.cpp

```diff
@@ -0,0 +1,35 @@
+#include "polynomial.h"
+#include "add.h"
+
+namespace SymEngine {
+
+    Polynomial::Polynomial(const std::string& var, map_uint_mpz&& dict) :
```

Comment:
Since we are not doing much insertion/deletions I think unordered_map is the way to go. But still let's keep this for now, once we have benchmarks we can try both, choose and adjust accordingly.